tests/components/matter/test_select.py

```diff
@@ -144,3 +145,47 @@ async def test_list_select_entities(
     await trigger_subscription_callback(hass, matter_client)
     state = hass.states.get("select.laundrywasher_temperature_level")
     assert state.state == "unknown"
+
+    # SpinSpeedCurrent
+    state = hass.states.get("select.laundrywasher_spin_speed")
+    assert state
+    assert state.state == "Off"
+    assert state.attributes["options"] == ["Off", "Low", "Medium", "High"]
+    set_node_attribute(matter_node, 1, 83, 1, 3)
+    await trigger_subscription_callback(hass, matter_client)
+    state = hass.states.get("select.laundrywasher_spin_speed")
+    assert state.state == "High"
+    # test select option
+    await hass.services.async_call(
+        "select",
+        "select_option",
+        {
+            "entity_id": "select.laundrywasher_spin_speed",
+            "option": "High",
+        },
+        blocking=True,
+    )
+    assert matter_client.write_attribute.call_count == 1
+    assert matter_client.write_attribute.call_args == call(
+        node_id=matter_node.node_id,
+        attribute_path=create_attribute_path_from_attribute(
+            endpoint_id=1,
+            attribute=clusters.LaundryWasherControls.Attributes.SpinSpeedCurrent,
+        ),
+        value=3,
+    )
+    # test that an invalid value (e.g. 253) leads to an unknown state
+    set_node_attribute(matter_node, 1, 83, 1, 253)
+    await trigger_subscription_callback(hass, matter_client)
+    state = hass.states.get("select.laundrywasher_spin_speed")
+    assert state.state == "unknown"
```

Comment:
Let's move testing the DynamicList variant of the enum to its own dedicated test function to separate the tests per variant of each platform better. Its not needed perse but makes it better readable/maintainable in the long run.